### PR TITLE
Log JDK boot modules on startup.

### PR DESCRIPTION
**Pull request by mbien**
could be useful while inspecting user logs (e.g https://github.com/apache/netbeans/issues/8026) since it would identify headless JDKs or JREs with non-standard module lists.

```
...
  Java Boot Modules       = [java.base, java.compiler, java.datatransfer, java.desktop, java.instrument, java.logging, java.management, java.management.rmi, java.naming, java.net.http, java.prefs, java.rmi, java.scripting, java.security.jgss, java.security.sasl, java.smartcardio, java.sql, java.sql.rowset, java.transaction.xa, java.xml, java.xml.crypto, jdk.accessibility, jdk.attach, jdk.charsets, jdk.compiler, jdk.crypto.cryptoki, jdk.dynalink, jdk.editpad, jdk.httpserver, jdk.internal.ed, jdk.internal.jvmstat, jdk.internal.le, jdk.internal.md, jdk.internal.opt, jdk.jartool, jdk.javadoc, jdk.jconsole, jdk.jdeps, jdk.jdi, jdk.jdwp.agent, jdk.jfr, jdk.jlink, jdk.jpackage, jdk.jshell, jdk.jsobject, jdk.jstatd, jdk.localedata, jdk.management, jdk.management.agent, jdk.management.jfr, jdk.naming.dns, jdk.naming.rmi, jdk.net, jdk.nio.mapmode, jdk.sctp, jdk.security.auth, jdk.security.jgss, jdk.unsupported, jdk.unsupported.desktop, jdk.xml.dom, jdk.zipfs]
...
```

### CODE REVIEW

File: platform/core.startup/src/org/netbeans/core/startup/TopLogging.java [hunk 1]
```diff
@@ -309,6 +313,32 @@ private static String createBootClassPath() {
...
+        try {
+            Class<?> mod_class = Class.forName("java.lang.Module");
+            Class<?> ml_class = Class.forName("java.lang.ModuleLayer");
+            @SuppressWarnings("unchecked")
+            Set<?> mods = (Set<?>)ml_class.getDeclaredMethod("modules").invoke(
+                ml_class.getDeclaredMethod("boot").invoke(null)
+            );
+            return mods.stream().map(mod -> {
+                try {
+                    return (String) mod_class.getMethod("getName").invoke(mod);
```
Comment (matthiasblaesing) [inline platform/core.startup/src/org/netbeans/core/startup/TopLogging.java:331]: You could pull the method lookup out of the loop.

**Comment by matthiasblaesing** [comment]
Apart from the small suggestion, this makes sense to me.

File: platform/core.startup/src/org/netbeans/core/startup/TopLogging.java (hunk 1 above)
Comment (mbien) [reply platform/core.startup/src/org/netbeans/core/startup/TopLogging.java:331]: oh you are right! will update

**Review by matthiasblaesing — APPROVED** [review]
Thank you

**Pull request merged**